### PR TITLE
add dragPoint call to zgui.plot

### LINKED ISSUE #328: zgui: dragpoints support
Hi, love this project!

Are there any plans to add `DragPoints` to the Implot wrapper?

Thanks!

### PR DESCRIPTION
**Pull request by ssteinbach**
Closes #328

Thanks for the great library!

I'm not really sure about the color parameter?  I also only exposed the `dragPoint` function + flags, not the other "drag tool" things.

### CODE REVIEW

File: libs/zgui/src/plot.zig [hunk 1]
```diff
@@ -537,6 +537,32 @@ extern fn zguiPlot_PlotShaded(
...
+    no_cursors: bool = false,
+    no_fit: bool = false,
+    no_no_inputs: bool = false,
+    delayed: bool = false,
+    _padding: u28 = 0,
+};
+const DragPointArgs = struct {
+    x: *f64,
+    y: *f64,
+    col: *const [4]f32,
```
Comment (michal-z) [inline libs/zgui/src/plot.zig:550]:
```suggestion
    col: [4]f32,
```

File: libs/zgui/src/plot.zig [hunk 2]
```diff
@@ -537,6 +537,32 @@ extern fn zguiPlot_PlotShaded(
...
 ) void;
 
+pub const DragToolFlags = packed struct(u32) {
+    no_cursors: bool = false,
+    no_fit: bool = false,
+    no_no_inputs: bool = false,
+    delayed: bool = false,
+    _padding: u28 = 0,
+};
+const DragPointArgs = struct {
```
Comment (michal-z) [inline libs/zgui/src/plot.zig:547]:
```suggestion
const DragPoint = struct {
```
Let's change the name for consistency.

**Review by michal-z — CHANGES_REQUESTED** [review]

**Comment by michal-z** [comment]
Thanks!

File: libs/zgui/src/plot.zig [hunk 3]
```diff
@@ -537,6 +537,32 @@ extern fn zguiPlot_PlotShaded(
...
+    x: *f64,
+    y: *f64,
+    col: [4]f32,
+    size: f32 = 4,
+    flags: DragToolFlags = .{},
+};
+pub fn dragPoint(id: i32, args: DragPointArgs) bool {
```
Comment (ssteinbach) [inline libs/zgui/src/plot.zig:554]:
```suggestion
pub fn dragPoint(id: i32, args: DragPoint) bool {
```

**Comment by ssteinbach** [comment]
Doh!  Slightly fast - I needed to apply the other two changes but was on my phone.  I'll make a PR.  Give me 5 minutes!

File: libs/zgui/src/plot.zig (hunk 2 above)
Comment (ssteinbach) [reply libs/zgui/src/plot.zig:547]: resolved in #332

**Comment by ssteinbach** [comment]
see #332 for the other fix

**Pull request merged**